src/plugins/intel_npu/tests/unit/npuw/permute.cpp

```diff
@@ -0,0 +1,255 @@
+            if (out_idx % 2 == 0) {
+                output[out_byte_idx] = (output[out_byte_idx] & 0xF0) | (in_val & 0x0F);
+            } else {
+                output[out_byte_idx] = (output[out_byte_idx] & 0x0F) | ((in_val & 0x0F) << 4);
+            }
+        } else {
+            output[out_idx] = input[in_idx];
+        }
+    }
+}
+
+}  // namespace details
+
+namespace {
+
+void PermuteTestsBase::make_input() {
+    size_t nElements = shape_size(input_shape);
+
+    ASSERT_EQ((type.bitwidth() * nElements) % 8, 0)
+        << "Input len has to be byte boundary aligned, but was " << type.bitwidth() * nElements << " bits";
+    const size_t nBytes = type.bitwidth() * nElements / 8;
+
+    input.resize(nBytes);
+    ref_output.resize(nBytes);
+    output.resize(nBytes);
+
+    std::fill(ref_output.begin(), ref_output.end(), 0);
+    std::fill(output.begin(), output.end(), 0);
+
+    std::mt19937 rng(42);  // Fixed seed for reproducibility
+    std::uniform_int_distribution<int> dist(0, 100);
+
+    for (size_t i = 0; i < nBytes; ++i) {
+        input[i] = static_cast<int8_t>(dist(rng));
+    }
+
+    inTensor = ov::Tensor(type, input_shape, input.data());
+}
+
+void PermuteTestsBase::SetUp(const PermuteTestsParams& getParam) {
+    ShapesInitializer shapeInit;
+
+    std::tie(type, shapeInit, axes) = getParam;
+
+    std::vector<int> input;
+    shapeInit(input);
+
+    input_shape = ov::Shape{input.begin(), input.end()};
+
+    make_input();
+
+    make_ref_output();
+}
+
+std::string PermuteTestsBase::ToString() const {
+    std::ostringstream result;
+    result << (isNegative() ? "NEGATIVE_" : "") << "[";
+
+    for (size_t i = 0; i != input_shape.size(); i++) {
+        result << input_shape[i] << ((i + 1 == input_shape.size()) ? "" : "x");
+    }
+    result << "]"
+           << "_type_" << type;
+
+    result << "_axis";
+
+    for (size_t i = 0; i < axes.size(); i++) {
+        result << "_" << axes[i];
+    }
+
+    return result.str();
+}
+
+/**
+ * Negative test cases has to be carefully reviewed, to still remain positive runs at some points
+ * @return
+ */
+bool PermuteTestsBase::isNegative() const {
+    return false;
+}
+
+void PermuteTestsBase::make_ref_output() {
+    auto rows = input_shape[0] * input_shape[1];
+    auto cols = input_shape[2];
+
+    std::vector<size_t> dims(input_shape.begin(), input_shape.end());
+    if (type == ov::element::f16) {
+        details::permute<uint16_t>(reinterpret_cast<const uint16_t*>(input.data()),
+                                   reinterpret_cast<uint16_t*>(ref_output.data()),
+                                   dims,
+                                   axes);
+    } else if (type == ov::element::f32) {
+        details::permute<float>(reinterpret_cast<const float*>(input.data()),
+                                reinterpret_cast<float*>(ref_output.data()),
+                                dims,
+                                axes);
+    } else {
```

Comment:
Let's check for exact type instead of just `else`:

```c++
} else if (type == ov::element::i4) {
   ...
} else {
// Fail somehow (.e.g ASSERT_TRUE(false);)
}

Reply:
Done.